grommet/grommet
# Added onChild function on Carousel component

#### What does this PR do?

This PR adds functionality to the **Carousel** component.  Indeed, it was not possible to retrieve the index of the active image from the Carousel. This is why I added a function called **onChild** which allows you to retrieve this index as soon as it changes.

#### Where should the reviewer start?

As this is not a visual feature, he will have to do what is described in the section _How should this be manually tested_.

#### What testing has been done on this PR?

The commit of my changes went without any error and all the tests passed successfully. I also tested my changes locally with a build of my changes. Everything works as I described above and below.

#### How should this be manually tested?

We need to add the onChild property on the component, and check if the index change in the console is the one we expect when we click on the buttons, or when the carousel is in play mode.

```
<Carousel
 controls="arrows"
 fill={true}
 initialChild={0}
 onChild={(index) =>
  console.log("Index of the element currently shown.", index)
 }
>
 {...}
</Carousel>
```

#### Any background context you want to provide?

I really need this feature for a project I've been working on for over a year now. But, I would also like to take advantage of other new features from grommet as time goes by, so I am submitting this PR.

#### What are the relevant issues?

In grommet v1, this feature was available. I don't know why it's not available in grommet v2. I'm sure that this feature can be practical and used by a lot of people.

#### Screenshots (if appropriate)

/

#### Do the grommet docs need to be updated?

I have already added the description of the new property in the `doc.js` file found in `src\js\components\Carousel`, and the `readme.md` file has been updated accordingly.
I have also run the **npm test -- -u** command to update the other files accordingly.
I don't know if that's enough for you.

#### Should this PR be mentioned in the release notes?

It's up to you.

#### Is this change backwards compatible or is it a breaking change?

This change is backwards compatible.
